tests/xcm-simulator/build.rs

```diff
@@ -0,0 +1,140 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+/// build.rs
+/// Compile and copy the contract artifacts to be used as fixture
+/// in tests
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use contract_build::{
+    BuildArtifacts, BuildMode, Features, ManifestPath, Network, OptimizationPasses, OutputType,
+    Target, UnstableFlags, Verbosity,
+};
+
+const DEFAULT_FIXTURES_DIR: &'static str = "./fixtures";
+const DEFAULT_CONTRACTS_DIR: &'static str = "./contracts";
+
+/// Execute the clousre with given directory as current dir
+fn with_directory<T, F: FnOnce() -> T>(dir: &Path, f: F) -> T {
```

Comment:
Surprised it doesn't exist already :upside_down_face:

Reply:
Not in `std`/`fs` atleast 🤷‍♂️